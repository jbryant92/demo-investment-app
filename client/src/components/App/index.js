import React from 'react';
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from 'react-router-dom';
import { Header } from 'semantic-ui-react';

import Campaigns from '../Campaigns';
import Campaign from '../Campaign';
import Investment from '../Investment';
import '../../App.css'

const App = () => (
  <div>
    <Header as='h1' className='app-header'>Demo investment app</Header>
    <Router>
      <Switch>
        <Route exact path="/" render={() => (
            <Redirect to="/campaigns"/>
        )}/>
        <Route exact path='/campaigns' component={ Campaigns } />
        <Route path='/campaigns/:id' component={ Campaign } />
        <Route path='/investments/:id' component={ Investment } />
        <Route render={() => <div>404</div>} />
      </Switch>
    </Router>
  </div>
)

export default App;