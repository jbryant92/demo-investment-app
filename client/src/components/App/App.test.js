import React from 'react';
import ReactDOM from 'react-dom';
import App from './index';
import enzymeToJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

it('renders without crashing', () => {
  const instance = shallow(
        <App />
    )
    expect(enzymeToJson(instance)).toMatchSnapshot()
});
