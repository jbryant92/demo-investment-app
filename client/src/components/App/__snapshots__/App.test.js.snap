// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div>
  <Header
    as="h1"
    className="app-header"
  >
    Demo investment app
  </Header>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        path="/campaigns/:id"
      />
      <Route
        component={[Function]}
        path="/investments/:id"
      />
      <Route
        render={[Function]}
      />
    </Switch>
  </BrowserRouter>
</div>
`;
